### ISSUE TITLE
Non-terminating character macro for #\- breaks #: reader.

### ISSUE THREAD

**Issue by karlosz** [Clozure/ccl#489]
The following minimal example shows a bug in CCL where defining a non-terminating character macro breaks symbol tokenization with #: (but not without #:!)

```
;; no-op, other than making #- a character macro.
(let ((standard-readtable (copy-readtable nil)))
  (set-macro-character #\- (lambda (stream char)
                             (unread-char char stream)
                             (let ((*readtable* standard-readtable))
                               (read stream t nil t)))
                       t)
  (read-from-string "-abcd") ; OK
  (read-from-string "#:-abcd") ; errors
  )
```

Note that the print name for '-abcd is also different after making #\- a character macro, though I think that's OK. Furthermore, this works correctly on sbcl, cmucl, clisp, and ecl.

I think the function `READ-SYMBOL-TOKEN` is at fault: it should do a check for non-terminating-ness for character macros.

**Comment by xrme** [comment]
#476 complains about the printer doing unnecessary slashification, by the way.

**Comment by xrme** [comment]
Fixed in f1f596337bb2ef67234e9ecbe822bde51fc6eea4.
